# 1장 1강: 가설검정 프레임워크의 이해 — 실습문제

## 실습 목표

- 분석 상황에 맞게 귀무가설(H₀)과 대립가설(H₁)을 설정할 수 있다.
- `scipy.stats.ttest_1samp()`로 단일표본 t검정을 수행할 수 있다.
- 유의수준과 p-value를 비교하여 귀무가설 기각 여부를 판단할 수 있다.
- 제1종 오류와 제2종 오류를 분석 상황에 맞게 설명할 수 있다.
- 연구 목적에 따라 양측검정과 단측검정을 구분할 수 있다.

## 실습 환경 / 데이터

- Python
- pandas
- scipy.stats
- `ames_housing.csv`

이번 실습에서는 미국 아이오와주 Ames 지역의 주택 자료를 사용합니다.

주요 컬럼은 다음과 같습니다.

| 컬럼 | 의미 |
|---|---|
| `SalePrice` | 주택 판매가격 |
| `GrLivArea` | 지상층 생활면적 |
| `OverallQual` | 주택의 전반적인 품질 점수(1~10) |
| `YearBuilt` | 건축연도 |

> 모든 검정의 유의수준은 별도 지시가 없으면 `α = 0.05`로 설정합니다.

## 실습 준비

1. 필요한 라이브러리를 불러오세요.
2. Ames Housing 데이터를 `df`에 불러오세요.
3. 데이터의 행과 열 개수, 컬럼명, 상위 5개 행을 확인하세요.

In [1]:
# 실습 준비 코드를 작성하세요.
from pathlib import Path
import pandas as pd
from scipy import stats

df = pd.read_csv("ames_housing.csv")

---

## 필수 1. 평균 판매가격에 대한 양측검정

### 문제 1-1. 평균 판매가격이 180,000달러와 다른가?

#### 문제 설명

Ames 지역의 주택 평균 판매가격에 대한 기존 기준이 **180,000달러**라고 가정합니다.  
제공된 표본의 평균 판매가격이 이 기준과 다른지 단일표본 t검정으로 확인하세요.

#### 요구사항

1. `SalePrice`에서 결측치를 제거하여 `sale_price`에 저장하세요.
2. 표본 수와 표본평균을 출력하세요.
3. 다음 가설을 코드 주석이나 마크다운으로 작성하세요.
   - H₀: Ames 지역의 평균 판매가격은 180,000달러이다.
   - H₁: Ames 지역의 평균 판매가격은 180,000달러가 아니다.
4. 유의수준 `alpha`를 0.05로 설정하세요.
5. `stats.ttest_1samp()`와 `alternative="two-sided"`를 사용해 양측검정을 수행하세요.
6. t통계량과 p-value를 소수점 넷째 자리까지 출력하세요.
7. p-value와 유의수준을 비교해 귀무가설 기각 여부를 출력하세요.

#### 해석 질문

**Q1.** 이 문제에서 양측검정을 사용하는 이유는 무엇인가요?  
**Q2.** 검정 결과에 따라 귀무가설을 기각할 수 있나요?  
**Q3.** 귀무가설을 기각하지 못했다면 평균 판매가격이 정확히 180,000달러라고 증명된 것인가요?

#### 제출 결과

- 가설 설정
- 표본 수와 표본평균
- 단일표본 t검정 코드
- t통계량과 p-value
- 귀무가설 판단
- Q1~Q3 답변

In [2]:
# 필수 1 코드를 작성하세요.
sale_price = df["SalePrice"].dropna()
mu0 = 180000
alpha = 0.05

result = stats.ttest_1samp(
    sale_price,
    popmean=mu0,
    alternative="two-sided"
)

print(f"표본 수: {len(sale_price)}")
print(f"표본평균: {sale_price.mean():,.2f}달러")
print(f"통계량: {result.statistic:.4f}")
print(f"p-value: {result.pvalue:.4f}")

if result.pvalue <= alpha:
    print("판단: 귀무가설을 기각합니다.")
else:
    print("판단: 귀무가설을 기각하지 못합니다.")

표본 수: 1460
표본평균: 180,921.20달러
통계량: 0.4431
p-value: 0.6578
판단: 귀무가설을 기각하지 못합니다.


### 필수 1 답변 작성란

- **Q1.** 
-> 평균이 기준보다 높은지 낮은지 방향성을 물어보는 것이 아니라 180,000달러와 다른지? 를 묻는 문제이기 때문.
- **Q2.**
-> 아니요 귀무가설을 기각하지 못합니다. p-value가 약 0.6578로 유의수준 0.05보다 크다.
- **Q3.**
-> 귀무가설이 참이라고 증명할 근거를 찾은 것이 아님

---

## 필수 2. 양측검정과 단측검정 비교

### 문제 2-1. 평균 품질 점수는 6점과 다른가, 그리고 6점보다 높은가?

#### 문제 설명

주택의 전반적인 품질을 나타내는 `OverallQual`의 기준 점수를 **6점**으로 설정했습니다.  
같은 표본으로 다음 두 연구 질문을 각각 검정하고 결과를 비교하세요.

- 질문 A: 평균 품질 점수가 6점과 **다른가?**
- 질문 B: 평균 품질 점수가 6점보다 **높은가?**

#### 요구사항

1. `OverallQual`에서 결측치를 제거하여 `overall_qual`에 저장하세요.
2. 표본 수와 표본평균을 출력하세요.
3. 질문 A의 가설을 작성하고 `alternative="two-sided"`로 양측검정을 수행하세요.
4. 질문 B의 가설을 작성하고 `alternative="greater"`로 단측검정을 수행하세요.
5. 두 검정의 t통계량과 p-value를 출력하세요.
6. 유의수준 0.05를 기준으로 각 검정의 귀무가설 기각 여부를 출력하세요.
7. 같은 데이터인데 두 p-value가 다르게 계산되는 이유를 설명하세요.

#### 해석 질문

**Q1.** 질문 A와 질문 B의 대립가설은 어떻게 다른가요?  
**Q2.** 양측검정과 단측검정의 결론은 각각 무엇인가요?  
**Q3.** 분석 결과를 먼저 확인한 뒤 p-value를 작게 만들기 위해 단측검정으로 바꿔도 되나요?  
**Q4.** 실제 평균 품질 점수가 6점인데도 검정 결과가 유의하다고 판단했다면 어떤 오류인가요?

#### 제출 결과

- 두 검정의 가설
- 표본 수와 표본평균
- 양측검정 및 단측검정 코드
- 두 검정의 t통계량과 p-value
- 검정별 귀무가설 판단
- Q1~Q4 답변

In [3]:
# 필수 2 코드를 작성하세요.

import pandas as pd
from scipy import stats

# 1. 결측치 제거
overall_qual = df["OverallQual"].dropna()

# 2. 표본 수, 표본평균
print("표본 수:", len(overall_qual))
print("표본평균:", overall_qual.mean())

mu = 6  # 문제에서 주어진 기준값으로 바꾸세요

# 3. 양측검정  H0: mu = 6  /  H1: mu != 6
t_two, p_two = stats.ttest_1samp(overall_qual, mu, alternative="two-sided")

# 4. 단측검정  H0: mu <= 6  /  H1: mu > 6
t_one, p_one = stats.ttest_1samp(overall_qual, mu, alternative="greater")

# 5. 출력
print(f"양측: t={t_two:.4f}, p={p_two:.4f}")
print(f"단측: t={t_one:.4f}, p={p_one:.4f}")

# 6. 기각 여부
alpha = 0.05
print("양측 귀무가설 기각" if p_two < alpha else "양측 귀무가설 기각 못함")
print("단측 귀무가설 기각" if p_one < alpha else "단측 귀무가설 기각 못함")

표본 수: 1460
표본평균: 6.0993150684931505
양측: t=2.7439, p=0.0061
단측: t=2.7439, p=0.0031
양측 귀무가설 기각
단측 귀무가설 기각


### 필수 2 답변 작성란

- **Q1.**     
-> 질문 A의 대립가설은 평균은 6점과 다르다 라는 개념으로 양/음의 방향을 정하지 않음
-> 질문 B의 대립가설은 평균 > 6으로 양의 방향으로 명확히 확인하려고 함
- **Q2.**
-> 양측검정에서는 평균 품질 점수가 6점과 다르다고 판단
-> 단측검정에서는 평균 품질 점수가 6점보다 높다고 판단
- **Q3.**
-> 안 된다. 검정 방향은 데이터를 확인하기 전에 연구 목적과 근거에 따라 정해야한다.
-> 결과를 본 뒤에 방향성을 바꾸면 제 1종 오류 위험을 통제하기 어렵다.

- **Q4.**
-> 실제 귀무가설이 참인데 귀무가설을 기각했으므로 제 1종 오류이다.
-> 실제 평균이 6점인데 표본을 보고 6점과 다르다고 잘못 판단한 경우


---

## 과제 1. 평가 문항 기반 독립 과제

### 문제 3-1. 평균 지상층 생활면적이 1,550보다 작은가?

#### 문제 설명

한 부동산 분석가는 Ames 지역 주택의 평균 지상층 생활면적이 **1,550보다 작다**고 주장합니다.  
`GrLivArea`를 사용하여 이 주장을 단일표본 t검정으로 확인하세요.

> 이 문제는 필수 문제에서 연습한 가설 설정과 단측검정 절차를 새로운 변수에 적용하는 문제입니다.

#### 요구사항

1. `GrLivArea`에서 결측치를 제거하여 분석 표본을 준비하세요.
2. 표본 수와 표본평균을 확인하세요.
3. 연구 주장에 맞는 귀무가설과 대립가설을 작성하세요.
4. 양측검정과 단측검정 중 적절한 검정 방향을 선택하고 그 이유를 설명하세요.
5. 단일표본 t검정을 수행하세요.
6. t통계량과 p-value를 소수점 넷째 자리까지 출력하세요.
7. 유의수준 0.05를 기준으로 귀무가설 기각 여부를 판단하세요.
8. 분석가의 주장을 뒷받침할 통계적 근거가 있는지 해석하세요.

#### 해석 질문

**Q1.** 이 문제에서 적절한 `alternative` 값은 무엇인가요?  
**Q2.** p-value는 귀무가설이 참일 확률을 의미하나요?  
**Q3.** 실제 평균이 1,550보다 작은데 귀무가설을 기각하지 못했다면 어떤 오류인가요?

#### 제출 결과

- 가설 설정과 검정 방향 선택 이유
- 표본 수와 표본평균
- 검정 코드
- t통계량과 p-value
- 귀무가설 판단과 분석 결론
- Q1~Q3 답변

In [5]:
# 과제 1 코드를 작성하세요.

import pandas as pd
import numpy as np
from scipy import stats

df = pd.read_csv('ames_housing.csv')
area = df['GrLivArea'].dropna()

n = len(area)
mean = area.mean()
std = area.std(ddof=1)
se = std / np.sqrt(n)

print(f"표본 수(n)     : {n}")
print(f"표본평균       : {mean:.4f}")
print(f"표본표준편차   : {std:.4f}")
print(f"표준오차       : {se:.4f}")

mu0 = 1550
t_stat, p_two = stats.ttest_1samp(area, mu0, alternative='less')
_, p_two = stats.ttest_1samp(area, mu0, alternative='two-sided')

print(f"\nt 통계량       : {t_stat:.4f}")
print(f"p-value(단측)  : {p_one:.4f}")
print(f"p-value(양측)  : {p_two:.4f}")

alpha = 0.05
if p_one < alpha:
    print(f"\n판단: p({p_one:.4f}) < {alpha} -> 귀무가설 기각")
else:
    print(f"\n판단: p({p_one:.4f}) >= {alpha} -> 귀무가설 기각 실패")

ci = stats.t.interval(0.95, n - 1, loc=mean, scale=se)
print(f"95% 신뢰구간   : [{ci[0]:.2f}, {ci[1]:.2f}]")


표본 수(n)     : 1460
표본평균       : 1515.4637
표본표준편차   : 525.4804
표준오차       : 13.7525

t 통계량       : -2.5113
p-value(단측)  : 0.0061
p-value(양측)  : 0.0121

판단: p(0.0061) < 0.05 -> 귀무가설 기각
95% 신뢰구간   : [1488.49, 1542.44]


### 과제 1 답변 작성란

**Q1.** 이 문제에서 적절한 `alternative` 값은 무엇인가요? 
-> 'less'

**Q2.** p-value는 귀무가설이 참일 확률을 의미하나요?
-> p-value는 귀무가설이 참일 확률이 아니라, 귀무가설이 참이라는 가정 아래 관측 결과가 얼마나 드문지를 나타내는 값

**Q3.** 실제 평균이 1,550보다 작은데 귀무가설을 기각하지 못했다면 어떤 오류인가요?
-> 1종 오류

---

## 실습 마무리

아래 질문에 답하세요.

1. 이번 실습에서는 표본평균이 기준값과 다른지 판단하기 위해 어떤 검정을 사용했나요?
-> 단일표본 T검정, 하나의 표본을 이용하여 모집단 평균이 특정 기준값과 다른지
-> 또는 한쪽 방향으로 차이가 있는지 확인
-> 1장 1강 실습문제에서는 주택 판매가격 표본을 이용하여 평균이 180,000과 다른지 검정했다.

2. 귀무가설의 기각 여부는 무엇을 기준으로 판단했나요?
-> p-value의 p 값과 유의수준(0.05)를 비교
-> p-value <= 0.05이면 귀무가설을 기각하고, p-value > 0.05이면 기각하지 못함

3. 양측검정과 단측검정은 연구 질문에서 어떤 차이가 있나요?
-> 양측검정은 기준과 다른지 확인하고
-> 단측검정은 기준보다 크거나 작은 특정 방향을 확인한다.


4. 귀무가설을 기각하지 못한다는 것은 무엇을 의미하나요?
-> 귀무가설을 참이라고 증명한 것이 아니라 대립가설을 지지할 증거가 충분하지 않다는 의미입니다.

5. 제1종 오류와 제2종 오류는 각각 어떤 상황인가요?
-> 제1종 오류는 실제로 귀무가설이 참인데 이를 기각하는 오류
-> 제2종 오류는 실제로 귀무가설이 거짓인데 이를 기각하지 못하는 오류